ddtrace/llmobs/_experiment.py

```diff
@@ -74,6 +100,14 @@ def push(self) -> None:
         new_version = self._dne_client.dataset_batch_update(self._id, self._records)
         self._version = new_version
 
+    @overload
+    def __getitem__(self, index: int) -> DatasetRecord:
```

Comment:
🤔 why the addition for these two overloads? the method below should handle the union, no?

Reply:
type hint checks complained about accessing dataset[idx], since this could theoretically return a `Union[DatasetRecord, List[DatasetRecord]]`. I could either just type ignore that line (smelly) or just explicitly define this overload method 😢

Reply:
ahh yeah makes sense!